holmes/plugins/toolsets/grafana/loki/toolset_grafana_loki.py

```diff
@@ -85,28 +121,38 @@ def _invoke(self, params: dict, context: ToolInvokeContext) -> StructuredToolRes
         )
 
         config = self.toolset._grafana_config
+        query_str = params.get("query", '{query="no_query_fallback"}')
         try:
             data = execute_loki_query(
                 base_url=get_base_url(config),
                 api_key=config.api_key,
                 headers=config.headers,
-                query=params.get(
-                    "query", '{query="no_query_fallback"}'
-                ),  # make sure a string returns. fall back to query that will return nothing.
+                query=query_str,
                 start=start,
                 end=end,
                 limit=params.get("limit") or DEFAULT_LOG_LIMIT,
             )
+
+            explore_url = _build_grafana_loki_explore_url(
```

Comment:
Does it work for both cases when you use Loki directly or when using with grafana proxy ( the datasource_uid case)? 
There is some hidden logic in base_url=get_base_url(config) ( which is bad) but handles url for both these cases?

Just wondering